qiskit/quantum_info/operators/measures.py

```diff
@@ -350,6 +350,89 @@ def cvx_bmat(mat_r, mat_i):
     return sol
 
 
+def unitary_diamond_distance(channel1: Operator, channel2: Operator) -> float:
```

Comment:
From a quick look at `_input_formatter`, I think the type annotation can be generalized to `Operator | QuantumChannel`, right?

Reply:
Hmm, I'm not super familiar with the `QuantumChannel` data type. If the channel is unitary, will running `channel.data` always return an n by n unitary matrix? Do I have to add 
```
if isinstance(op1, QuantumChannel):
            op1  = op1.to_operator()
```
Or does everything already work nicely

Reply:
Hmm on further thought let's just keep it to Operator because that's the class meant to store unitary objects. Maybe add QuantumCircuit, but I don't feel strongly about it.